aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -197,7 +197,10 @@ void reduction_out_mps(
              (dtype.value() == kFloat || dtype.value() == kHalf || dtype.value() == kInt)) {
             inputCastDtype = getMPSDataType(dtype.value());
           } else if (input_type != MPSDataTypeInt32   &&
-                     input_type != MPSDataTypeFloat32) {
+                     input_type != MPSDataTypeFloat32 &&
```

Comment:
Instead of repeating this check everywhere, lets use this [function.](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/mps/OperationUtils.mm#L57)

Reply:
Lets followup the cleanup in a separate PR.

Reply:
We'll cleanup and refactor in a separate PR later.